toolkit/imageconfigs/additionalconfigs/cloud-init.cfg

```diff
@@ -0,0 +1,120 @@
+# The top level settings are used as module
+# and system configuration.
+
+# A set of users which may be applied and/or used by various modules
+# when a 'default' entry is found it will reference the 'default_user'
+# from the distro configuration specified below
+users:
+   - default
+   - name: root
+     lock-passwd: false
+
+# If this is set, 'root' will not be able to ssh in and they 
+# will get a message to login instead as the above $user (ubuntu)
+disable_root: false
+
+#Vmware guest customization.
+disable_vmware_customization: true
+
+# This will cause the set+update hostname module to not operate (if true)
+preserve_hostname: false
+
+# datasources to check for cloud-config
+datasource_list: [ 
+                  NoCloud, 
+                  ConfigDrive,
+                  OVF,
+                  OpenStack,
+#                  OpenNebula, 
+#                  AltCloud, 
+#                  MAAS, 
+#                  GCE, 
+#                  CloudSigma, 
+#                  Ec2, 
+#                  CloudStack,
+#                  VmxGuestinfo, 
+                  None,
+                  Azure
+                 ]
+
+# Example datasource config
+# datasource: 
+#    Ec2: 
+#      metadata_urls: [ 'blah.com' ]
+#      timeout: 5 # (defaults to 50 seconds)
+#      max_wait: 10 # (defaults to 120 seconds)
+
+# The modules that run in the 'init' stage
+cloud_init_modules:
+# - migrator
+# - seed_random
+ - bootcmd
+ - write-files
+ - growpart
+ - resizefs
+ - disk_setup
+ - mounts
+ - set_hostname
+ - update_hostname
+ - update_etc_hosts
+# - ca-certs
+ - rsyslog
+ - users-groups
+ - ssh
+
+# The modules that run in the 'config' stage
+cloud_config_modules:
+# Emit the cloud config ready event
+# this can be used by upstart jobs for 'start on cloud-config'.
+# - emit_upstart
+# - disk_setup
+# - mounts
+ - ssh-import-id
+ - set-passwords
+ - package-update-upgrade-install
+# - landscape
+# - timezone
+# - puppet
+# - chef
+# - salt-minion
+# - mcollective
+# - disable-ec2-metadata
+ - runcmd
+# - byobu
+ - yum-add-repo
+
+# The modules that run in the 'final' stage
+cloud_final_modules:
+# - rightscale_userdata
+ - scripts-vendor
+ - scripts-per-once
+ - scripts-per-boot
+ - scripts-per-instance
+ - scripts-user
+ - ssh-authkey-fingerprints
+# - keys-to-console
+# - phone-home
+ - final-message
+# - power-state-change
```

Comment:
Clean up the commented out options in this section and the ones above it.

Reply:
Done